nixos/modules/services/network-filesystems/openafs/server.nix

```diff
@@ -296,13 +400,16 @@ in {
         after = [ "network.target" ];
         wantedBy = [ "multi-user.target" ];
         restartIfChanged = false;
-        unitConfig.ConditionPathExists = [
+        unitConfig.ConditionPathExists = mkIf (cfg.authentication.keytab == null) [
           "|/etc/openafs/server/KeyFileExt"
         ];
         preStart = ''
           mkdir -m 0755 -p /var/openafs
           ${optionalString (netInfo != null) "cp ${netInfo} /var/openafs/netInfo"}
           ${optionalString useBuCellServDB "cp ${buCellServDB}"}
+        '' + lib.optionalString (cfg.authentication.keytab != null) ''
+          rm -f /etc/openafs/server/KeyFileExt
+          ${asetkeyScript}
```

Comment:
I just noticed that `asetkeyScript` never calls `asetkey` when the configured principal is not in the keytable or the keytable cannot be read. Thus, `KeyFileExt` never gets created. Please, make `preStart` fail with a descriptive error message in that case.